providers/account.rb

```diff
@@ -169,7 +169,24 @@ def home_ssh_dir_resource(exec_action)
 
 def authorized_keys_resource(exec_action)
   # avoid variable scoping issues in resource block
-  ssh_keys = Array(new_resource.ssh_keys)
+  ssh_keys = []
+  Array(new_resource.ssh_keys).each do |item|
+    # if item is a valid ssh key, copy it.
+    if item =~ /^(ssh-(dss|rsa|ed25519)|ecdsa-sha2-\w+) AAAA/
```

Comment:
I'm personally guilty of abusing this resource to add comments to my authorized keys file. Can you please add something similar to `(\s+)?#`. I'm thinking it'd be good not to chomp comments even if it has some erroneous whitespace at the front.